### PR TITLE
Update Zoo.cs - Exception handling

### PR DESCRIPTION
**Pull request by CCianfloneDev**
Added documentation and Impl. exception handling. For Issue #12

### CODE REVIEW

File: ZooKeepingSystem/Zoo.cs [hunk 1]
```diff
@@ -38,19 +50,38 @@ public string ZooName
...
             AnimalsInZoo = animals;
         }
 
         /// <summary>
-        /// Initializes a zoo with a name, 0 cages, 1 zoo keeper, and no animals in the care.
+        /// Initializes a zoo with a name, 1 cage, 1 zoo keeper, and no animals in the care.
         /// </summary>
         /// <param name="zooName">The name of the zoo.</param>
-        public Zoo(string zooName) : this(zooName, 0, 1, null)
+        public Zoo(string zooName) : this(zooName, 1, 1, null)
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] Add exception documentation from invoking `this(zooName, 1, 1, null)`

File: ZooKeepingSystem/Zoo.cs [hunk 2]
```diff
@@ -38,19 +50,38 @@ public string ZooName
...
+                throw new ArgumentNullException("Name", "Name cannot be null.");
+            }
+            if (numberOfCages < 0)
+            {
+                throw new ArgumentOutOfRangeException("NumberOfCages", "Argument cannot be less than zero.");
+            }
+            if (numberOfZooKeepers < 0)
+            {
+                throw new ArgumentOutOfRangeException("NumberOfZooKeepers", "Argument cannot be less than zero.");
+            }
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] The logic for state validation should be in the property set accessors.

File: ZooKeepingSystem/Zoo.cs [hunk 3]
```diff
@@ -38,19 +50,38 @@ public string ZooName
...
+        /// Thrown when initialized with no name.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when number of cages or number of zoo keepers is less than zero.
+        /// </exception>
         public Zoo(string zooName, int numberOfCages, int numberOfZooKeepers, Animals animals)
         {
+            if (zooName == null)
+            {
+                throw new ArgumentNullException("Name", "Name cannot be null.");
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] Change param name from `Name` to `zooName`
- [x] zooName also should not be blank i.e. `""`

File: ZooKeepingSystem/Zoo.cs [hunk 4]
```diff
@@ -38,19 +50,38 @@ public string ZooName
...
+            }
+            if (numberOfCages < 0)
+            {
+                throw new ArgumentOutOfRangeException("NumberOfCages", "Argument cannot be less than zero.");
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] Change param name to camel case

File: ZooKeepingSystem/Zoo.cs [hunk 5]
```diff
@@ -38,19 +50,38 @@ public string ZooName
...
+            }
+            if (numberOfZooKeepers < 0)
+            {
+                throw new ArgumentOutOfRangeException("NumberOfZooKeepers", "Argument cannot be less than zero.");
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] Change NumberOfZooKeepers to camel case

File: ZooKeepingSystem/Zoo.cs [hunk 6]
```diff
@@ -7,28 +7,40 @@ namespace ZooKeepingSystem
...
+        /// Gets and set animals in the zoo.
+        /// </summary>
         public Animals AnimalsInZoo
         {
             get;
             set;
         }
 
+        /// <summary>
+        /// Gets and sets the name of the zoo.
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] Doesn't have a `public set` so should be documented as a `get` only

**Review by DZ-FSDev — CHANGES_REQUESTED** [review]
**CR for #20**

File: ZooKeepingSystem/Zoo.cs [hunk 7]
```diff
@@ -7,22 +7,46 @@ namespace ZooKeepingSystem
...
-            set;
+            get
+            {
+                return this.numberOfCages;
+            }
+            set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException("numberOfCages", "Argument cannot be less than zero.");
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] The argument referenced in property setters is usually `value`

File: ZooKeepingSystem/Zoo.cs [hunk 8]
```diff
@@ -7,22 +7,46 @@ namespace ZooKeepingSystem
...
-            set;
+            get
+            {
+                return this.numberOfZooKeepers;
+            }
+            set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException("numberOfZooKeepers", "Argument cannot be less than zero.");
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] The argument referenced in property setters is usually `value`

File: ZooKeepingSystem/Zoo.cs [hunk 9]
```diff
@@ -35,12 +59,22 @@ public Animals AnimalsInZoo
...
-            private set;
+            get
+            {
+                return this.zooName;
+            }
+            private set
+            {
+                if (String.IsNullOrEmpty(value))
+                {
+                    throw new ArgumentNullException("zooName", "Name cannot be null.");
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] The argument referenced in property setters is usually `value`

File: ZooKeepingSystem/Zoo.cs [hunk 10]
```diff
@@ -58,29 +92,22 @@ public string ZooName
...
-
             ZooName = zooName;
             NumberOfCages = numberOfCages;
             NumberOfZooKeepers = numberOfZooKeepers;
             AnimalsInZoo = animals;
         }
 
         /// <summary>
-        /// Initializes a zoo with a name, 1 cage, 1 zoo keeper, and no animals in the care.
+        /// Initializes a zoo with a name, 1 cages, 1 zoo keeper, and no animals in the care.
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] 1 cage.

File: ZooKeepingSystem/Zoo.cs [hunk 11]
```diff
@@ -58,29 +92,22 @@ public string ZooName
...
         /// </summary>
         /// <param name="zooName">The name of the zoo.</param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when initialized with no name.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when number of cages or number of zoo keepers is less than zero.
+        /// </exception>
```
Comment (DZ-FSDev) [inline ZooKeepingSystem/Zoo.cs]: - [x] This exception shouldn't be thrown by this constructor.

**Review by DZ-FSDev — CHANGES_REQUESTED** [review]
**CR for #20**

**Review by DZ-FSDev — APPROVED** [review]
Looks good!

**Comment by CCianfloneDev** [comment]
Exception handl. successfully impl.
All menus are functioning as intended.

**Comment by CCianfloneDev** [comment]
Accidentally closed PR before merging, I reopened then merged, lol.

**Pull request merged**